.azure-pipelines/main-pipeline.yml

```diff
@@ -116,15 +116,27 @@ stages:
               vsixFile: '$(Pipeline.Workspace)/$(artifactPath)/$(packagedTask)'
               extensionTag: '$(devExtensionTag)'
               extensionName: '$(extensionName) DEV'
-              extensionVersion: '$(queryDevVersion.Extension.Version)'
-              updateTasksId: true
               extensionVisibility: 'private'
               extensionPricing: 'free'
               shareWith: 'datadog-ci'
+              extensionVersion: '$(queryDevVersion.Extension.Version)'
+              updateTasksId: true # The DEV task ID is overridden to 99cf84cd-0117-58b7-b1dd-d95e5ae9baf8 (based on the `extensionTag`)
+              updateTasksVersion: true # Do not require updating the task version during development
+              updateTasksVersionType: patch
+
+          - task: Bash@3
+            displayName: Wait for DEV extension availability
```

Comment:
I first tried to simplify things by removing the E2E pipeline and move its content in the main pipeline.

- Polling and running the E2E tests in the same `PublishDev` job didn't work
- Polling in `PublishDev` and running the E2E tests in a separate job, all of this in the same stage didn't work
- Polling in `PublishDev` and running the E2E tests in a separate stage didn't work

So we indeed still require a separate E2E **pipeline**. And now it works! We can run the E2E tests on the right version: the one that was just published.

Reply:
I guess moving this task at the beginning of the e2e pipeline won't allow to to resolve the latest published version in the following task of the e2e pipeline, right?
Could we add a comment about this, because it's not intuitive.